src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Services/TagHelperSemanticRangeVisitor.cs

```diff
@@ -384,9 +384,25 @@ private void AddSemanticRange(SyntaxNode node, int semanticKind)
             var range = node.GetRange(source);
 
             // LSP spec forbids multi-line tokens, so we need to split this up.
-            // Thankfully all instances of this have multiple component tokens.
             if (range.Start.Line != range.End.Line)
             {
+                var childNodes = node.ChildNodes();
+                if (childNodes.Count == 0)
+                {
+                    var content = node.GetContent();
```

Comment:
I believe this will "ToString" every inner-node which means if you comment out an entire file (definitely reasonable) you'd make a hugeeeeeee string.  I'd recommend iterating the tokens specifically and finding the newline tokens

Reply:
The problem is that Razor comment contents are not tokenized by line like other strings (otherwise my previous "multiline" handling system would have taken care of this).

The alternative here would be to alter the compiler to tokenize each line of a Razor comment, consume that change and I think it'd fix itself (though we should still add tests). I think that's actually the better fix since I have some concerns about trying to parse elements of a SyntaxTree like I do here. I've not touched the Razor compiler before but I'd be interested to dig in a bit.

Reply:
> I've not touched the Razor compiler before but I'd be interested to dig in a bit.

Totally up to you if you're interested! Probably the hardest part would be updating the [actual syntax representation](https://github.com/dotnet/aspnetcore/blob/2dced017cbb4c2c236ecdc9a568b4ef8bc6b6550/src/Razor/Microsoft.AspNetCore.Razor.Language/src/Syntax/Syntax.xml#L23-L25) other then that you'd need to update the [creation of the updated comment syntax kind](https://github.com/dotnet/aspnetcore/blob/c925f99cddac0df90ed0bc4a07ecda6b054a0b02/src/Razor/Microsoft.AspNetCore.Razor.Language/src/Legacy/TokenizerBackedParser.cs#L367) and the [tokenization of the comments](https://github.com/dotnet/aspnetcore/blob/c925f99cddac0df90ed0bc4a07ecda6b054a0b02/src/Razor/Microsoft.AspNetCore.Razor.Language/src/Legacy/Tokenizer.cs#L283)